OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDiscussionTopicsFragment.java

```diff
@@ -141,10 +138,12 @@ protected void onResponse(@NonNull final CourseTopics courseTopics) {
                 ArrayList<DiscussionTopic> allTopics = new ArrayList<>();
                 allTopics.addAll(courseTopics.getNonCoursewareTopics());
                 allTopics.addAll(courseTopics.getCoursewareTopics());
-
                 List<DiscussionTopicDepth> allTopicsWithDepth = DiscussionTopicDepth.createFromDiscussionTopics(allTopics);
-                discussionTopicsAdapter.setItems(allTopicsWithDepth);
-                discussionTopicsAdapter.notifyDataSetChanged();
+                topicsList.clear();
+                topicsList.addAll(getTopicsHeaders());
+                topicsList.addAll(allTopicsWithDepth);
+                adapter.submitList(topicsList);
+                adapter.notifyItemRangeInserted(0, topicsList.size());
```

Comment:
No need to notify after submitList function. The function itself notify the list.